pkg/cli/codearea.go

```diff
@@ -196,6 +200,56 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a non-word abbreviation.
+func (w *codeArea) expandAbbr() {
+	var abbr, full string
+	// Try to expand an abbreviation, preferring the longest one
+	w.Abbreviations(func(a, f string) bool {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+		return false // check the next abbr
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr() {
+	w.StateMutex.Lock()
+	defer w.StateMutex.Unlock()
+
+	ilen := len(w.inserts)
+	w.WordAbbreviations(func(a, f string) bool {
```

Comment:
Small word expansion should prefer the longest match too, so this should first walk through all pairs, find the longest match, and then do the expansion.

There is no need to make the callback return a `bool` after you make this change.

Reply:
Due to restricting matches to word boundaries there can never be more than a single match.